### PR TITLE
Multi Device Support

### PR DESCRIPTION
**Pull request by gmbnt**

### CODE REVIEW

File: src/org/thoughtcrime/securesms/loki/DeviceLinkingDialogDelegate.kt [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+import org.whispersystems.signalservice.loki.api.PairingAuthorisation
+
 interface DeviceLinkingDialogDelegate {
+    fun handleDeviceLinkAuthorized() {}
+    fun handleDeviceLinkingDialogDismissed() {}
+}
 
-    fun handleDeviceLinkAuthorized() // TODO: Device link
-    fun handleDeviceLinkingDialogDismissed()
+interface DeviceLinkingViewDelegate: DeviceLinkingDialogDelegate {
```
Comment (gmbnt) [inline src/org/thoughtcrime/securesms/loki/DeviceLinkingDialogDelegate.kt]: ?????

Comment (gmbnt) [reply src/org/thoughtcrime/securesms/loki/DeviceLinkingDialogDelegate.kt]: This setup is now really weird

File: src/org/thoughtcrime/securesms/loki/SeedActivity.kt [hunk 2]
```diff
@@ -44,11 +53,18 @@ class SeedActivity : BaseActionBarActivity() {
...
-        toggleModeButton.setOnClickListener { toggleMode() }
+        toggleRestoreModeButton.setOnClickListener { mode = Mode.Restore }
+        toggleRegisterModeButton.setOnClickListener { mode = Mode.Register }
+        toggleLinkModeButton.setOnClickListener { mode = Mode.Link }
         registerOrRestoreButton.setOnClickListener { registerOrRestore() }
         Analytics.shared.track("Seed Screen Viewed")
     }
+
+    override fun onDestroy() {
+        super.onDestroy()
```
Comment (gmbnt) [inline src/org/thoughtcrime/securesms/loki/SeedActivity.kt]: ?

File: src/org/thoughtcrime/securesms/loki/SeedActivity.kt [hunk 3]
```diff
@@ -44,11 +53,18 @@ class SeedActivity : BaseActionBarActivity() {
         setContentView(R.layout.activity_seed)
         setUpLanguageFileDirectory()
         updateSeed()
+        updateUI()
         copyButton.setOnClickListener { copy() }
-        toggleModeButton.setOnClickListener { toggleMode() }
+        toggleRestoreModeButton.setOnClickListener { mode = Mode.Restore }
+        toggleRegisterModeButton.setOnClickListener { mode = Mode.Register }
+        toggleLinkModeButton.setOnClickListener { mode = Mode.Link }
         registerOrRestoreButton.setOnClickListener { registerOrRestore() }
```
Comment (gmbnt) [inline src/org/thoughtcrime/securesms/loki/SeedActivity.kt]: This does more than register or restore now

File: src/org/thoughtcrime/securesms/loki/SeedActivity.kt (hunk 2 above)
Comment (Mikunj) [reply src/org/thoughtcrime/securesms/loki/SeedActivity.kt]: woops, this was when i was still thinking about where to put the logic, we can remove this.

**Pull request merged**